plugins/action_update_review_status/src/updateReviewsStatuses.ts

```diff
@@ -0,0 +1,184 @@
+      // Get section codes of the changed responses
+      const changedSections = await db.getChangedSections(
+        changedResponses.map(({ templateElementId }) => templateElementId)
+      )
+
+      // Filter out reviews that are not level 1 and don't intersect with
+      // changed section codes
+      const reviewsToUpdate = reviews.filter(
+        ({ reviewStatus, levelNumber, assignedSections }) =>
+          levelNumber === 1 &&
+          reviewStatus === ReviewStatus.Submitted &&
+          assignedSections.some((sectionCode) => changedSections.includes(sectionCode))
+      )
+
+      // Set remaining reviews to "PENDING" (using the "changeStatus" action)
+      const results: Promise<ActionPluginOutput>[] = []
+      reviewsToUpdate.forEach((review) =>
+        results.push(
+          changeStatus({
+            parameters: {
+              newStatus: ReviewStatus.Pending,
+              reviewId: review.reviewId,
+              isReview: true,
+            },
+            applicationData,
+            DBConnect,
+          })
+        )
+      )
+
+      const reviewStatuses = await Promise.all(results)
+
+      return {
+        status: ActionQueueStatus.Success,
+        error_log: '',
+        output: {
+          updatedReviews: reviewStatuses.map(({ output }) => output),
+        },
+      }
+    } else {
+      // REVIEW SUBMISSIONS:
+      if (!reviewId) throw new Error('Missing reviewId')
+
+      const thisReviewLevel = reviews.find((review) => review.reviewId === reviewId)?.levelNumber
+
+      if (!thisReviewLevel) throw new Error('Invalid reviewId')
+
+      const nextLevelReview = reviews.filter(
+        ({ levelNumber }) => levelNumber === thisReviewLevel + 1
+      )
+
+      // Set review status to "PENDING" for next-level reviews
+      const results: Promise<ActionPluginOutput>[] = []
+      nextLevelReview.forEach((review) =>
+        results.push(
+          changeStatus({
+            parameters: {
+              newStatus: ReviewStatus.Pending,
+              reviewId: review.reviewId,
+              isReview: true,
+            },
+            applicationData,
+            DBConnect,
+          })
+        )
+      )
+
+      if (thisReviewLevel > 1) {
+        // Set lower-level review to "CHANGES REQUESTED" if the current review
+        // has disagreed with any response decisions:
+
+        const disagreedSections = await db.getChangedSections(
+          (changedResponses as ChangedReviewResponse[])
+            .filter(
+              ({ reviewResponseDecision }) =>
+                reviewResponseDecision === ReviewResponseDecision.Disagree
+            )
+            .map(({ templateElementId }) => templateElementId)
+        )
+
+        const lowerReviewsToUpdate = reviews.filter(
+          ({ levelNumber, assignedSections }) =>
+            levelNumber < thisReviewLevel &&
+            assignedSections.some((section) => disagreedSections.includes(section))
+        )
```

Comment:
This is updating ALL lower-level reviews. Should it just be level 1?

Reply:
No. As per my previous comment on the documentation.

Reply:
Also - I believe the same rule applies here. We shouldn't account for reviews that have status set to DISCONTINUED, since those are no longer the current ones... And therefor shouldn't be updated.